sdk/formrecognizer/azure-ai-formrecognizer-perf/src/main/java/com/azure/ai/formrecognizer/perf/ReceiptRecognitionStreamTest.java

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.ai.formrecognizer.perf;
+
+import com.azure.ai.formrecognizer.models.FormRecognizerOperationResult;
+import com.azure.ai.formrecognizer.models.RecognizedForm;
+import com.azure.ai.formrecognizer.perf.core.ServiceTest;
+import com.azure.ai.formrecognizer.perf.core.Utility;
+import com.azure.core.util.polling.SyncPoller;
+import com.azure.perf.test.core.PerfStressOptions;
+import reactor.core.publisher.Mono;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.List;
+
+public class ReceiptRecognitionStreamTest extends ServiceTest<PerfStressOptions> {
+    private FileInputStream fileInputStream;
+    private String receiptFilePath = "src/main/resources/samplesFiles/contoso-receipt.png";
+    private long fileLength;
+    public ReceiptRecognitionStreamTest(PerfStressOptions options) {
+        super(options);
+    }
+
+    @Override
+    public Mono<Void> globalSetupAsync() {
+        return super.globalSetupAsync().then(createReceiptInputStream());
+    }
+
+    @Override
+    public Mono<Void> globalCleanupAsync() {
+        return closeReceiptInputStream().then(super.globalCleanupAsync());
+    }
+
+    @Override
+    public void run() {
+        formRecognizerClient.beginRecognizeReceipts(fileInputStream, fileLength).waitForCompletion();
+    }
+
+    @Override
+    public Mono<Void> runAsync() {
+        SyncPoller<FormRecognizerOperationResult, List<RecognizedForm>> syncPoller =
+        formRecognizerAsyncClient.beginRecognizeReceipts(Utility.toFluxByteBuffer(fileInputStream), fileLength)
```

Comment:
Can we initialize the `Flux<ByteBuffer>` globally instead of per test, this will reduce noise on functionality we aren't trying to benchmark (at least not in this instance). If you do this you'll want to add a `.map(ByteBuffer::duplicate)` as sharing the `Flux` may lead to `ByteBuffers` reaching their terminal read state and subsequent tests failing.

Reply:
Currently, `Utility.toFluxByteBuffer(fileInputStream)` is being invoked on each single invocation of this method within a single thread. This will heavily impact the performance.

Create the FluxByteBuffer outside as class variable in the class instructor if needed per thread. 
If you want to make it global then make its class instance variable static as well and set it up in the globalSetup method.